### ISSUE TITLE
Staging metadata temp files occurs after job directory cleanup

### ISSUE THREAD

**Issue by natefoo** [galaxyproject/pulsar#111]
For example (previous log messages trimmed, but it successfully stages out the output datasets, metadata pickles, metrics, etc.):

```pytb
pulsar.client.staging.down DEBUG 2016-04-21 17:38:49,659 collecting dynamic output_metadata file __instrument_core_epoch_end
pulsar.client.staging.down INFO 2016-04-21 17:38:49,659 collecting output __instrument_core_epoch_end with action FileAction[action_type=copy]
pulsar.client.client DEBUG 2016-04-21 17:38:49,660 Copying path [/jetstream/tacc-scratch0/main/jobs/12555432/metadata/__instrument_core_epoch_end] to [/galaxy-repl/main/jobdir/012/555/12555432/__instrument_core_epoch_end]
pulsar.client.staging.down DEBUG 2016-04-21 17:38:49,668 Cleaning up job (failed [False], cleanup_job [onsuccess])
galaxy.model.metadata DEBUG 2016-04-21 17:38:52,263 loading metadata from file for: HistoryDatasetAssociation 24473418
galaxy.objectstore DEBUG 2016-04-21 17:38:52,341 Using preferred backend 'corral-repl' for creation of MetadataFile 1066560
galaxy.objectstore CRITICAL 2016-04-21 17:38:52,354 Error copying /jetstream/tacc-scratch0/main/jobs/12555432/metadata/metadata_temp_file_WTcLXh to /galaxy-repl/main/files/_metadata_files/001/066/metadata_1066560.dat: [Errno 2] No such file or directory: u'/jetstream/tacc-scratch0/main/jobs/12555432/metadata/metadata_temp_file_WTcLXh'
raven.base.Client DEBUG 2016-04-21 17:38:52,360 Sending message of length 615 to https://sentry.galaxyproject.org/api/2/store/
galaxy.jobs.runners.pulsar ERROR 2016-04-21 17:38:52,363 Job wrapper finish method failed
Traceback (most recent call last):
  File "/cvmfs/main.galaxyproject.org/galaxy/lib/galaxy/jobs/runners/pulsar.py", line 485, in finish_job
    remote_working_directory=remote_working_directory
  File "/cvmfs/main.galaxyproject.org/galaxy/lib/galaxy/jobs/__init__.py", line 1269, in finish
    dataset.metadata.from_JSON_dict( output_filename, path_rewriter=path_rewriter )
  File "/cvmfs/main.galaxyproject.org/galaxy/lib/galaxy/model/metadata.py", line 167, in from_JSON_dict
    dataset._metadata[ name ] = param.from_external_value( external_value, dataset, **from_ext_kwds )
  File "/cvmfs/main.galaxyproject.org/galaxy/lib/galaxy/model/metadata.py", line 603, in from_external_value
    alt_name=os.path.basename(mf.file_name) )
  File "/cvmfs/main.galaxyproject.org/galaxy/lib/galaxy/objectstore/__init__.py", line 494, in update_from_file
    return self._call_method('update_from_file', obj, ObjectNotFound, True, **kwargs)
  File "/cvmfs/main.galaxyproject.org/galaxy/lib/galaxy/objectstore/__init__.py", line 631, in _call_method
    return self.backends[object_store_id].__getattribute__(method)(obj, **kwargs)
  File "/cvmfs/main.galaxyproject.org/galaxy/lib/galaxy/objectstore/__init__.py", line 422, in update_from_file
    raise ex
IOError: [Errno 2] No such file or directory: u'/jetstream/tacc-scratch0/main/jobs/12555432/metadata/metadata_temp_file_WTcLXh'
```

**Comment by natefoo** [comment]
Correction... the metadata file is staged out. What happens is that the `metadata_out_HistoryDatasetAssociation_<id>_....` generated on the node contains the absolute path to the metadata temp file:

```json
"bam_index": {"kwds": {}, "__class__": "MetadataTempFile", "filename": "/jetstream/scratch0/test/jobs/542041/metadata/metadata_temp_file_L7hpRC"}
```

Pulsar correctly stages this back to Galaxy's job directory, but because the absolute path is read from the metadata output file, Galaxy tries to read from the wrong file (which has been removed, or in the case that there's not a shared filesystem, the path does not exist).

**Comment by natefoo** [comment]
This was fixed by @jmchilton in galaxyproject/galaxy#2384, thanks!
